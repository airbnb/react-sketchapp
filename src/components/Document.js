/* @flow */
import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.node,
};

class Document extends React.Component {
  render() {
    return <document>{this.props.children}</document>;
  }
}

Document.propTypes = propTypes;

module.exports = Document;
