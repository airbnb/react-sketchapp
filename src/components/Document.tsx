import * as React from 'react';
import * as PropTypes from 'prop-types';

export const DocumentPropTypes = {
  children: PropTypes.node,
};

type Props = PropTypes.InferProps<typeof DocumentPropTypes>;

export class Document extends React.Component<Props> {
  static propTypes = {
    children: PropTypes.node,
  };

  render() {
    return <sketch_document>{this.props.children}</sketch_document>;
  }
}
