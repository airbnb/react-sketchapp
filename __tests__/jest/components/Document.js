import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Document from '../../../src/components/Document';

describe('<Document />', () => {
  it('renders children', () => {
    const tree = renderer
      .create(
        <Document>
          <foo>
            <bar />
          </foo>
        </Document>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
