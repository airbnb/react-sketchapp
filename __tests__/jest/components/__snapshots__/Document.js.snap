// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Document /> renders children 1`] = `
<document>
  <foo>
    <bar />
  </foo>
</document>
`;
